src/controllers/activegate/reconciler/automaticapimonitoring/reconciler.go

```diff
@@ -0,0 +1,88 @@
+package automaticapimonitoring
+
+import (
+	"fmt"
+
+	"github.com/Dynatrace/dynatrace-operator/src/controllers/dynakube/status"
+	"github.com/Dynatrace/dynatrace-operator/src/dtclient"
+	"github.com/pkg/errors"
+)
+
+type Reconciler struct {
+	dtclient.Client
+}
+
+func NewReconciler(clt dtclient.Client) *Reconciler {
+	return &Reconciler{
+		clt,
+	}
+}
+
+func (r *Reconciler) Reconcile(dkState *status.DynakubeState) error {
+	name := dkState.Instance.Name
+	kubeSystemUUID := dkState.Instance.Status.KubeSystemUUID
+	objectID, err := r.ensureSettingExists(r.Client, name, kubeSystemUUID)
+
+	if err != nil {
+		return err
+	}
+
+	if objectID != "" {
+		log.Info(fmt.Sprintf("created setting '%s' for Cluster '%s'. Settings object ID: %s", name, kubeSystemUUID, objectID))
+	} else {
+		log.Info(fmt.Sprintf("setting '%s' for Cluster '%s' already exists.", name, kubeSystemUUID))
+	}
+
+	return nil
+}
+
+func (r *Reconciler) ensureSettingExists(dtc dtclient.Client, name string, kubeSystemUUID string) (string, error) {
```

Comment:
no need to pass `dtclient.Client`, can be accessed by calling `r.dtc` (after name change in struct)